pkg/scheduler/controller/shoot/scheduler_control.go

```diff
@@ -218,6 +247,21 @@ func filterSeedsMatchingSeedSelector(cloudProfile *gardencorev1beta1.CloudProfil
 	return matchingSeeds, nil
 }
 
+func filterSeedsMatchingProviders(cloudProfile *gardencorev1beta1.CloudProfile, shoot *gardencorev1beta1.Shoot, seedList []*gardencorev1beta1.Seed) ([]*gardencorev1beta1.Seed, error) {
+	var possibleProviders []string
+	if cloudProfile.Spec.SeedSelector != nil {
```

Comment:
same thing here, can we not just return all seeds if  cloudProfile.Spec.SeedSelector.Providers empty?

Reply:
if possibleProviders does not contain anything, there is still a filtering, it selects only seeds with a provider matching the shoots provider